python/cmsis_svd/parser.py

```diff
@@ -164,38 +163,6 @@ class SVDParser(object):
     def for_xml_file(cls, path, remove_reserved=False):
         return cls(ET.parse(path), remove_reserved)
 
-    @classmethod
-    def for_packaged_svd(cls, vendor, filename, remove_reserved=False):
-        resource = "data/{vendor}/{filename}".format(
-            vendor=vendor,
-            filename=filename
-        )
-
-        filename = pkg_resources.resource_filename("cmsis_svd", resource)
-        return cls.for_xml_file(filename, remove_reserved)
-
-    @classmethod
-    def for_mcu(cls, mcu):
-        mcu = mcu.lower()
-        vendors = pkg_resources.resource_listdir("cmsis_svd", "data")
-        for vendor in vendors:
-            fnames = pkg_resources.resource_listdir("cmsis_svd", "data/%s" % vendor)
-            for fname in fnames:
-                filename = fname.lower()
-                if not filename.endswith(".svd"):
-                    continue
-                filename = filename[:-4]
-                if mcu.startswith(filename):
-                    return cls.for_packaged_svd(vendor, fname)
-            for fname in fnames:
-                filename = fname.lower()
-                if not filename.endswith(".svd"):
-                    continue
-                filename = "^%s.*" % filename[:-4].replace('x', '.')
-                if re.match(filename, mcu):
-                    return cls.for_packaged_svd(vendor, fname)
-        return None
-
```

Comment:
I definitely like removing the svd files from this package, and thus these functions becomes obsolete.

But however, I think it makes sense, even though this package isn't version 1 yet, it is still kind of adopted, and makes sense to break the API in helpful way. For example, semantic versioning states that minor versions should be backwards compatible.

So I suggest instead of removing them entirely, keep them as stubs, but let them always fail in a helpful way, like raising an exception with a message telling that the feature of bundled svd files is removed. At least during a transition period, possibly up until 1.0.0. It will at least give the users of the tools that relies on this package to understand what the reason is.

Reply:
Another option would be to use some env var/default dir where files can be placed and have these function fall back there. That way you could set `SVD_PATH=/usr/share/cmsis-svd/data:/usr/lib/python3/dist-packages/cmsis-svd:~/.config/cmsis-svd/data` (which could be the implied default on Linux) and these functions could just continue to work as expected.

While the main package should not include the SVD files, it could still be helpful to allow for some per-vendor packages (or other simple means) to acquire these files as needed.

Reply:
@BenBE I am agree with your last note, or maybe just  add a root directory path as parameter of the both functions.

Reply:
> But however, I think it makes sense, even though this package isn't version 1 yet, it is still kind of adopted, and makes sense to break the API in helpful way. For example, semantic versioning states that minor versions should be backwards compatible.

For versions before 1.0 each "minor" version is equivalent in semantics to a post 1.0 major version.  So the move to 0.5 is allowed to break APIs in the same way as any major version and there's likely a slew of changes since the ancient 0.4 that are already semver breaking changes.

I believe it is important while we have some engagement to get some update to pypi and probably iterate a few releases before aiming for a 1.0.

Reply:
> While the main package should not include the SVD files, it could still be helpful to allow for some per-vendor packages (or other simple means) to acquire these files as needed.

I'm not opposed to this but would like to get 0.5 released in a basic form prior to that.  We can include in our releases an archive of the data and move toward a means of using that as well as CMSIS PACK files (which are just zipfiles with some definition of structure).

Reply:
Hi, I think, this is better to not release data in the python package anymore. I propose to produce a tarball as a release in another repository of this organisation, and pass the data folder as a git submodule.

Reply:
I propose this code as replacement, it works.

```python
    @classmethod
    def for_packaged_svd(cls, package_root, vendor, filename, remove_reserved=False):
        path = os.path.abspath(os.path.join(package_root, vendor, filename))
        return cls.for_xml_file(path, remove_reserved)

    @classmethod
    def for_mcu(cls, package_root, mcu):
        path = os.path.abspath(package_root)
        mcu = mcu.lower()
        vendors = os.listdir(path)
        for vendor in vendors:
            fnames = os.listdir("{}{}{}".format(path, os.sep, vendor))
            for fname in fnames:
                filename = fname.lower()
                if not filename.endswith(".svd"):
                    continue
                filename = filename[:-4]
                if mcu.startswith(filename):
                    return cls.for_packaged_svd(package_root, vendor, fname)
            for fname in fnames:
                filename = fname.lower()
                if not filename.endswith(".svd"):
                    continue
                filename = "^%s.*" % filename[:-4].replace('x', '.')
                if re.match(filename, mcu):
                    return cls.for_packaged_svd(package_root, vendor, fname)
        return None
```

Reply:
LGTM.

Maybe have `package_root` as optional, last parameter. If set to `None` ask `get_default_package_root()` for the default path. That way this shouldn't even impact existing code too much …

Reply:
This is more a question about the goal of this tool.

Is it only to be an SVD parser, which takes a file and outputs a model of the file. Or is it also an SVD directory, that also adds a file structure to it?

If this is going to be also a directory of SVD, then I agree with @posborne that it PACK files makes sense, as discussed in #170

Even though I see how the `package_root` can be used by the implementor, I'm not sure about the value, since in the end it would equally easy be to implement a file browser and open the svd file directly in my opinion.

Having it referring to the root of cmsis-svd would not be relevant since we're removing the svd file archive within cmsis-svd,

Reply:
@pengi I think this Python package must be only a SVD parser. I understand your feedback about the simplicity of impleenting a file browser. But I think provide these helpers to load svd file which will appreciate by end-user, thand write boiler plate code before using the lib, this is just more sexy to use. I understand that opinions can differ.

Also, In the next steps, I think it can be interesting to distribute or retrieve SVD files ( via CMSIS-Packs). Maybe with a downloader in this package or another, in the same spirit of the sub project [cmsis-pack-manager](https://pyocd.io/cmsis-pack-manager/) of Pyocd you point in the #170. But in a different way, since the goal of this project is to provide full access of the CMSIS-SVD specifications, than retrieve only partial data as explain in the cmsis-pack-manager's doc.